### PR TITLE
Pr/clipboard debug

### PR DESCRIPTION
**Pull request by uli42**
lots of improvements to clipboard, mostly debugging.

### CODE REVIEW

File: nx-X11/programs/Xserver/hw/nxagent/NXproperty.c [hunk 1]
```diff
@@ -93,9 +93,9 @@ nxagentWMStateRec;
 #undef  TEST
 #undef  DEBUG
 
-int 
+int
 ProcChangeProperty(ClientPtr client)
-{	      
+{
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/NXproperty.c]: Why would we want to diverge hw/nxagent/NXproperty.c and dix/proporty.c white-space wise. Is this a good idea?

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c [hunk 2]
```diff
@@ -163,6 +163,8 @@ const char * GetClientSelectionStageString(int stage)
 #define PrintClientSelectionStage()
 #endif
 
+#define SAFE_XFree(what) do {if (what) {XFree(what); what = NULL;}} while (0)
+
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: Shouldn't we define this much more globally, so that it can be reused at various other places?

Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: @Ionic uses this macro in his Xinerama tiling branch, too.

**Review by sunweaver — CHANGES_REQUESTED** [review]
Please check my comments. Currently, Github has out-of-sync webmirrors, so commenting on code ttw is really difficult and tricky.

About the FIXME regarding the reset of "i = 0" in one of the commits (Github would not let me comment inline): Not necessarily do we need to reset the "i" counter, currently it seems that the counter "i" is used to iterate over all selections, first up to nxagentCurrentSelection, then up to nxagentMaxSelections. This could be wanted. However, I am unsure and would have to investigate the clipboard code in more depths.

**Comment by uli42** [comment]
I have moved the macro to Utils.h, but please note that it is  a
different one from Ionic's (which calls free()  while mine calls
XFree())

You are also right about the whitespace stuff, I did not have in mind
that NXproperty.c is derived from Xorg code in dix. So I reverted that
one.

**Pull request merged**